### PR TITLE
bpo-46285: Add command-line option -p/--protocol to module http.server

### PR DESCRIPTION
**Pull request by geryogam**
This PR will add command-line option `-p`/`--protocol` to module [`http.server`](https://docs.python.org/3/library/http.server.html) which specifies the HTTP version to which the server is conformant (HTTP/1.1 servers can now be run from the command-line interface of module `http.server`):

> python -m http.server --protocol HTTP/1.1

https://bugs.python.org/issue46285

### CODE REVIEW

**Comment by merwok** [comment]
Thanks for the the PR!  Let’s merge this one after the bug fix to the same module.

**Review by JelleZijlstra — APPROVED** [review]

File: Doc/library/http.server.rst [hunk 1]
```diff
@@ -444,6 +444,15 @@ the following command uses a specific directory::
 .. versionadded:: 3.7
     ``--directory`` argument was introduced.
 
+By default, the server supports HTTP up to version HTTP/1.0. The option ``-p/--protocol``
```
Comment (merwok) [inline Doc/library/http.server.rst]: Should this say «up to»?  HTTP/0.9 had no headers I think, so a server with HTTP/1.0 really needs clients that support 1.0.   (I haven’t checked the code recently)

Comment (geryogam) [reply Doc/library/http.server.rst]: @merwok `BaseHTTPRequestHandler` does support HTTP/0.9, even when you configure it to be conformant with HTTP/1.0 or HTTP/1.1. But I agree that ‘up to’ was unnecessary. I have just updated the PR, thanks.

The relevant section on protocol versioning of the latest HTTP specification: https://httpwg.org/http-core/draft-ietf-httpbis-semantics-latest.html#protocol.version

File: Doc/library/http.server.rst [hunk 2]
```diff
@@ -193,7 +195,7 @@ provides three different variants:
 
    .. method:: handle_expect_100()
 
-      When a HTTP/1.1 compliant server receives an ``Expect: 100-continue``
+      When an HTTP/1.1 conformant server receives an ``Expect: 100-continue``
```
Comment (merwok) [inline Doc/library/http.server.rst:198]: `compliant` was correct, why change it?

Comment (geryogam) [reply Doc/library/http.server.rst:198]: @merwok Actually the two terms are not exact synonyms; ‘conformance’ is voluntary adherence, while ‘compliance’ is mandatory adherence:
https://www.kellertechnology.com/blog/compliance-vs-conformance-understanding-the-relationship-to-standard/

The term ‘conformant’ is the proper term in this context, used in the HTTP specification:
https://httpwg.org/http-core/draft-ietf-httpbis-semantics-latest.html#protocol.version

And the HTTP specification makes an explicit distinction between ‘conformant’ and ‘compliant’ here:
https://httpwg.org/http-core/draft-ietf-httpbis-semantics-latest.html#rfc.section.2.2.p.4):

> An implementation is considered conformant if it complies with all of the requirements associated with the roles it partakes in HTTP.

**Comment by maggyero** [comment]
> Please avoid rebases and force-pushes for CPython, github does not have a good experience for reviewers.

Okay, I was not aware of that.

File: Doc/library/http.server.rst (hunk 2 above)
Comment (maggyero) [reply Doc/library/http.server.rst:198]: What do you think?

Comment (geryogam) [reply Doc/library/http.server.rst:198]: Do you still prefer ‘compliant’? If that is what is preventing this PR from being merged, we can keep ‘compliant’.

Comment (merwok) [reply Doc/library/http.server.rst:198]: I haven’t had time to review the latest changes + check if the rest of the documentation uses compliant and/or conformant.
There is still time before 3.11 beta!

Comment (geryogam) [reply Doc/library/http.server.rst:198]: Yes there is still time. Should I add an entry in What’s new since it is a new feature?

Comment (merwok) [reply Doc/library/http.server.rst:198]: News fragment is enough!

Comment (geryogam) [reply Doc/library/http.server.rst:198]: Alright.

Comment (geryogam) [reply Doc/library/http.server.rst:198]: Hi @merwok. The 3.11 beta will be released next week. Could we merge?

**Review by JelleZijlstra — APPROVED** [review]

**Review by orsenthil — APPROVED** [review]
LGTM.

I had a brief hesitancy for exposing this protocol version via flag as the support is not comprehensive. I am pretty sure we are missing many features. 

But given that class supports `.protocol_version` as a attribute, exposing it via a command line flag is alright.

**Comment by maggyero** [comment]
Thanks @merwok, @JelleZijlstra, and @orsenthil for the review!

**Pull request merged**